### ISSUE TITLE
lua error

### ISSUE THREAD

**Issue by nukhub** [GurliGebis/WoWAddon-AngrierWorldQuests#59]
```
1x [ADDON_ACTION_BLOCKED] L'AddOn 'AngrierWorldQuests' a tenté d'appeler la fonction protégée 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:224: in function `CheckMouseButtonPassthrough'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:209: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:323: in function `AddWorldQuest'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:248: in function `RefreshAllData'
[string "@Blizzard_WorldMap/WM_WorldQuestDataProvider.lua"]:26: in function `RefreshAllData'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:180: in function `OnEvent'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:99: in function `SignalEvent'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:106: in function <...aceBlizzard_MapCanvas/Blizzard_MapCanvas.lua:105>
[string "=[C]"]: in function `secureexecuterange'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:109: in function `OnEvent'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:148: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:147>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
```

ty for your work

**Comment by Mythris29** [comment]
Same issue here

18x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:481: in function <!BugGrabber/BugGrabber.lua:481>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:189: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/QuestDataProvider.lua"]:228: in function `AddQuest'
[string "@Blizzard_SharedMapDataProviders/QuestDataProvider.lua"]:142: in function `CheckAddQuest'
[string "@Blizzard_SharedMapDataProviders/QuestDataProvider.lua"]:153: in function `RefreshAllData'
[string "@Blizzard_SharedMapDataProviders/QuestDataProvider.lua"]:106: in function `OnEvent'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:99: in function `SignalEvent'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:106: in function <...aceBlizzard_MapCanvas/Blizzard_MapCanvas.lua:105>
[string "=[C]"]: in function `secureexecuterange'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:109: in function `OnEvent'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:109: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:108>

Locals:
_ = Frame {

**Comment by ynazar1** [comment]
I'm not able to get that to trigger. 

Is there anything specific you're doing?

**Comment by nukhub** [comment]
was doing a world quest, i think it proc when i was in combat (im on 43.2 too btw)

**Comment by ynazar1** [comment]
FWIW: I did few world quests. No errors from AWQ. 

also Angrier World Quests does not have functionality of `SetPassThroughButtons` (i checked the code). This looks like lua taint. 

I actually get this instead:

```
1x [ADDON_ACTION_BLOCKED] AddOn 'TomCats' tried to call the protected function 'ObjectiveTrackerFrame:Show()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `Show'
[string "@Blizzard_ObjectiveTracker/Blizzard_ObjectiveTrackerContainer.lua"]:101: in function <...ectiveTracker/Blizzard_ObjectiveTrackerContainer.lua:43>
[string "=(tail call)"]: ?
[string "@Blizzard_SharedXML/MixinUtil.lua"]:341: in function <Blizzard_SharedXML/MixinUtil.lua:340>
``` 

Cautiously suggesting it's something else.

**Comment by ynazar1** [comment]
herebedragons-2.0 library has this piece of code which is more or less what everyone uses. 

herebedragons-Pins-2.0.lua
```
-- hack to avoid in-combat error on 10.1.5
worldmapProviderPin.SetPassThroughButtons = function() end
```
I think you got some ancient addon and ancient libraries that are taining the situation.

**Comment by ynazar1** [comment]
as an aside, i strongly suggest Visual Studio Code as it's great in finding all the files that use this function if you open the entire addons library with it.

**Comment by nukhub** [comment]
i did reproduce the error with only AWQ loaded, no other addons, doing a random quest

```
1x [ADDON_ACTION_BLOCKED] L'AddOn 'AngrierWorldQuests' a tenté d'appeler la fonction protégée 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:224: in function `CheckMouseButtonPassthrough'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:209: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/BonusObjectiveDataProvider.lua"]:97: in function <...haredMapDataProviders/BonusObjectiveDataProvider.lua:92>
[string "=[C]"]: in function `xpcall'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:76: in function `FireCallbacks'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:35: in function <.../Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua:31>
[string "=[C]"]: in function `accessor'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:52: in function `AddCallback'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:60: in function `AddCancelableCallback'
[string "@Blizzard_SharedMapDataProviders/BonusObjectiveDataProvider.lua"]:92: in function `RefreshAllData'
...
[string "=[C]"]: in function `Show'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:493: in function `SetUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:393: in function `ShowUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:174: in function <...ParentPanelManager/Mainline/UIParentPanelManager.lua:169>
[string "=[C]"]: in function `SetAttribute'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:888: in function `ShowUIPanel'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:106: in function `SetDisplayState'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:42: in function `HandleUserActionToggleSelf'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:511: in function `ToggleWorldMap'
[string "TOGGLEWORLDMAP"]:1: in function <[string "TOGGLEWORLDMAP"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
```

**Comment by ynazar1** [comment]
Super quick question, by chance are you already in pre-release access where exp works again?

**Comment by nukhub** [comment]
thank you

**Comment by ynazar1** [comment]
Slightly able to narrow it down. There are two tain errors. 

one is 
`Frame:SetPassThroughButtons()`
another is 
`Button:SetPassThroughButtons()`

Both are restricted actions in combat. 
The addon likely inherits old style templates when creating frames and buttons but i've not been able to get to the bottom of this

Definitely replicatable with combat even if AWQ is literally the only addon loaded.

**Comment by ynazar1** [comment]
@Shauren  FYI

**Comment by GurliGebis** [comment]
I just tried it myself - I cannot reproduce it (even with all my normal addons loaded).

Here is what I do test:

1: Pull something
2: Run this command to verify I'm in lockdown: `/run print(InCombatLockdown())` (it prints true)
3: I press L to open the map

No errors are logged in Bugsack - so for some reason, it doesn't break for me.

Is your testing methods identical to the above?

**Comment by nukhub** [comment]
`
[string "=[C]"]: in function `Show'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:493: in function `SetUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:336: in function `ShowUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:174: in function <...ParentPanelManager/Mainline/UIParentPanelManager.lua:169>
[string "=[C]"]: in function `SetAttribute'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:888: in function `ShowUIPanel'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:106: in function `SetDisplayState'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:42: in function `HandleUserActionToggleSelf'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:511: in function `ToggleWorldMap'
[string "TOGGLEWORLDMAP"]:1: in function <[string "TOGGLEWORLDMAP"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
`

another error, only AWQ loaded, v0.43.3, same thing (opening map while in combat)

**Comment by nukhub** [comment]
`2x [ADDON_ACTION_BLOCKED] L'AddOn 'AngrierWorldQuests' a tenté d'appeler la fonction protégée 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:224: in function `CheckMouseButtonPassthrough'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:209: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:323: in function `AddWorldQuest'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:248: in function `RefreshAllData'
[string "@Blizzard_WorldMap/WM_WorldQuestDataProvider.lua"]:26: in function `RefreshAllData'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:76: in function `OnMapChanged'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:595: in function <...aceBlizzard_MapCanvas/Blizzard_MapCanvas.lua:594>
[string "=[C]"]: in function `secureexecuterange'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:598: in function `OnMapChanged'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:303: in function `OnMapChanged'
...
[string "=[C]"]: in function `Show'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:493: in function `SetUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:336: in function `ShowUIPanel'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:174: in function <...ParentPanelManager/Mainline/UIParentPanelManager.lua:169>
[string "=[C]"]: in function `SetAttribute'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:888: in function `ShowUIPanel'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:106: in function `SetDisplayState'
[string "@Blizzard_WorldMap/QuestLogOwnerMixin.lua"]:42: in function `HandleUserActionToggleSelf'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:511: in function `ToggleWorldMap'
[string "TOGGLEWORLDMAP"]:1: in function <[string "TOGGLEWORLDMAP"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}`

another error, only AWQ loaded (v0.43.3), same thing : opening map while in combat

**Comment by ynazar1** [comment]
Still happenning

i think you got the frame but not the button 

1x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.

Addons loaded:
AWQ
Bugsack
BugGrabber

**Comment by ynazar1** [comment]
The error only happens first time after /reload in combat. Then it doesn't happen as it's now blocked by the internal stuff. 

My testing:
/reload
go fight closest bear...
press M (or L), this doesn't seem to make a difference
Bugsack gets error:
```
1x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:224: in function `CheckMouseButtonPassthrough'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:209: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:323: in function `AddWorldQuest'
[string "@Blizzard_SharedMapDataProviders/WorldQuestDataProvider.lua"]:248: in function `RefreshAllData'
[string "@Blizzard_WorldMap/WM_WorldQuestDataProvider.lua"]:26: in function `RefreshAllData'
```

![image](https://github.com/user-attachments/assets/63d89995-19c5-44c0-ac37-6e5f9ab0ac34)

**Comment by Shauren** [comment]
Taint happens here: https://github.com/GurliGebis/WoWAddon-AngrierWorldQuests/blob/cc22fbf4659a1abf606c6e488a8284e6fe94eac1/QuestFrame.lua#L1267 (this recursively taints all child tables and frames of `dataProvider`, including all existing AND FUTURE map pins)

I'm not sure this can be worked around without either gutting the addon functionality (hiding POIs) or entirely reimplementing all world quest POI and hiding default ones

**Comment by Shauren** [comment]
To expand on reproduction - this also happens when you switch zones on map while in combat

**Comment by ynazar1** [comment]
Perhaps looking at how the [World Quest List](https://www.curseforge.com/wow/addons/world-quests-list) does it may help. They pop up pins but no taint.

**Comment by ynazar1** [comment]
@Shauren how did you find exact place where taint happens?

**Comment by Shauren** [comment]
By looking at the stack trace of the error you can see that what is tainted are map pins

With that information you look at what AngrierWorldQuests does that is related to map pins and that is
* Hiding based on filters
* Highlighting when hovering over quest name in list

Then simply commenting out related code and testing ingame you will find that highlight doesn't trigger the error so it must have been the other pin interaction (indirect, via parent table `WorldMapFrame.dataProviders`)

**Comment by ynazar1** [comment]
Temp solution for now, just don't do anything while incombat:

Add to questframe.lua:
```
local function QuestFrame_Update()
	if InCombatLockdown() then
		return
	end
```
and
```
local function WorldMap_WorldQuestDataProviderMixin_ShouldShowQuest(self, info)
	if InCombatLockdown() then
		return
	end
```

**Comment by Shauren** [comment]
Taint happens when the function override happens, not inside it

**Comment by ynazar1** [comment]
Whelp I tried. Did i mention I hate how Lua frames work?

**Comment by Gnarfoz** [comment]
This particular annoyance can be squarely blamed on WoW's "taint" system. 🫤

**Comment by nukhub** [comment]
> I still find it quiet interresting that it doesn't happen on my machine. Only difference I can see must be some other addon I have loaded that does something, that causes this issue to not happen somehow. Once I get time (maybe monday or tuesday), I'll try and see if I can pin point which one it is (it that is the case at all) - if there is one, we should be able to figure out what needs to be done 🙂

this is my current addon list, if you want to rule them out from your machine

```
24/07/2024  08:20    <DIR>          !BugGrabber
15/08/2024  09:52    <DIR>          AdvancedInterfaceOptions
25/08/2024  13:09    <DIR>          AngrierWorldQuests
18/08/2024  11:11    <DIR>          AngryKeystones
18/01/2024  19:49    <DIR>          AnyFont
24/07/2024  08:20    <DIR>          Bartender4
24/08/2024  17:11    <DIR>          BartrubySummonPet
24/07/2024  08:20    <DIR>          BasicChatMods
25/08/2024  20:30    <DIR>          BigWigs
19/08/2024  11:55    <DIR>          BigWigs_Classic
25/08/2024  20:30    <DIR>          BigWigs_Core
20/08/2024  09:43    <DIR>          BigWigs_Dragonflight
25/08/2024  20:30    <DIR>          BigWigs_NerubarPalace
25/08/2024  20:30    <DIR>          BigWigs_Options
25/08/2024  20:30    <DIR>          BigWigs_Plugins
23/08/2024  08:05    <DIR>          BlacklistWarden
15/08/2024  09:52    <DIR>          BugSack
25/07/2024  19:38    <DIR>          ColorPickerPlus
18/01/2024  19:49    <DIR>          CooldownReadyFlashDisabler
18/01/2024  19:49    <DIR>          CrapAway
20/08/2024  09:43    <DIR>          Details
20/08/2024  09:43    <DIR>          Details_Compare2
20/08/2024  09:43    <DIR>          Details_DataStorage
20/08/2024  09:43    <DIR>          Details_EncounterDetails
20/08/2024  09:43    <DIR>          Details_RaidCheck
20/08/2024  09:43    <DIR>          Details_Streamer
20/08/2024  09:43    <DIR>          Details_TinyThreat
20/08/2024  09:43    <DIR>          Details_Vanguard
18/08/2024  11:11    <DIR>          DynamicCam
15/08/2024  09:52    <DIR>          EasyDelete
25/08/2024  13:09    <DIR>          ErrorFilter
20/08/2024  13:20    <DIR>          FastAutoRepair
22/08/2024  16:35    <DIR>          FasterLoot
20/08/2024  11:23    <DIR>          HideButtonGlow
21/08/2024  17:52    <DIR>          idTip
23/08/2024  08:05    <DIR>          ItemUpgradeQualityIcons
24/07/2024  08:20    <DIR>          Kui_Media
24/07/2024  08:20    <DIR>          Kui_Nameplates
24/07/2024  08:20    <DIR>          Kui_Nameplates_Core
24/07/2024  08:20    <DIR>          Kui_Nameplates_Core_Config
26/08/2024  13:26    <DIR>          Kui_Nameplates_TargetHelper
25/08/2024  20:30    <DIR>          LittleWigs
25/08/2024  20:30    <DIR>          LittleWigs_BattleForAzeroth
25/08/2024  20:30    <DIR>          LittleWigs_BurningCrusade
25/08/2024  20:30    <DIR>          LittleWigs_Cataclysm
25/08/2024  20:30    <DIR>          LittleWigs_Classic
25/08/2024  20:30    <DIR>          LittleWigs_Dragonflight
25/08/2024  20:30    <DIR>          LittleWigs_Legion
25/08/2024  20:30    <DIR>          LittleWigs_MistsOfPandaria
25/08/2024  20:30    <DIR>          LittleWigs_Shadowlands
25/08/2024  20:30    <DIR>          LittleWigs_WarlordsOfDraenor
25/08/2024  20:30    <DIR>          LittleWigs_WrathOfTheLichKing
20/08/2024  09:43    <DIR>          M6
15/08/2024  09:52    <DIR>          Masque
20/08/2024  14:14    <DIR>          MikScrollingBattleText
17/08/2024  19:12    <DIR>          MRT
20/08/2024  14:14    <DIR>          MSBTOptions
21/08/2024  09:31    <DIR>          MythicDungeonTools
13/08/2024  16:44    <DIR>          OmniCC
13/08/2024  16:44    <DIR>          OmniCC_Config
18/08/2024  11:11    <DIR>          OPie
15/08/2024  09:52    <DIR>          ParagonReputation
20/08/2024  09:43    <DIR>          PitBull4
20/08/2024  09:43    <DIR>          PitBull4_DogTagTexts
24/08/2024  08:26    <DIR>          Plumber
24/08/2024  08:26    <DIR>          PremadeGroupsFilter
23/08/2024  19:12    <DIR>          ProfessionShoppingList
26/08/2024  13:26    <DIR>          RaiderIO
26/08/2024  13:26    <DIR>          RaiderIO_DB_EU_F
26/08/2024  13:26    <DIR>          RaiderIO_DB_EU_M
26/08/2024  13:26    <DIR>          RaiderIO_DB_EU_R
26/08/2024  13:26    <DIR>          RaiderIO_DB_KR_F
26/08/2024  13:26    <DIR>          RaiderIO_DB_KR_M
26/08/2024  13:26    <DIR>          RaiderIO_DB_KR_R
26/08/2024  13:26    <DIR>          RaiderIO_DB_TW_F
26/08/2024  13:26    <DIR>          RaiderIO_DB_TW_M
26/08/2024  13:26    <DIR>          RaiderIO_DB_TW_R
26/08/2024  13:26    <DIR>          RaiderIO_DB_US_F
26/08/2024  13:26    <DIR>          RaiderIO_DB_US_M
26/08/2024  13:26    <DIR>          RaiderIO_DB_US_R
22/08/2024  23:44    <DIR>          RaidFrameSettings
15/08/2024  09:52    <DIR>          RCLootCouncil
27/07/2024  18:08    <DIR>          SexyMap
27/07/2024  18:08    <DIR>          SharedMedia
20/08/2024  13:20    <DIR>          SharedMedia_Causese
18/01/2024  19:49    <DIR>          SharedMedia_MyMedia
26/08/2024  13:26    <DIR>          SimpleItemLevel
19/08/2024  12:36    <DIR>          Simulationcraft
23/08/2024  09:35    <DIR>          TomTom
19/08/2024  12:36    <DIR>          TrufiGCD
26/08/2024  13:26    <DIR>          UnlimitedMapPinDistance
26/08/2024  13:26    <DIR>          WeakAuras
26/08/2024  13:26    <DIR>          WeakAurasArchive
26/08/2024  13:26    <DIR>          WeakAurasModelPaths
26/08/2024  13:26    <DIR>          WeakAurasOptions
26/08/2024  13:26    <DIR>          WeakAurasTemplates
```

**Comment by GurliGebis** [comment]
Thanks - I'll keep that list in mind 😊

**Comment by GurliGebis** [comment]
Okay, good news - I just had 10 minutes to do a test 🙂

If I load only AWQ, BugGrabber and BugSack - it fails the same way as described above.
If I load my normal set of addons, it doesn't.

This means two things.
1: We can fix it from code.
2: I should be able to find out which addon has the code that fixes the issue by bisecting my list of loaded addons.

I won't get time to do it tonight, but hopefully I get some time to work on it tomorrow.

**Comment by Shauren** [comment]
Big maybe #63

**Comment by ynazar1** [comment]
So I just looked around and there are many addons that are triggering this issue currently. More or less all addons that deal with map pins somehow trigger this sooner or later.

Was able to replicate passthroughbutton taint calls with (only addon enabled):
WorldQuestList
Gathermate2
SilverDragon_Overlays
AngrierWorldQuests

Few others i think too. 

TL:DR: Seems like this is an endemic issue for pretty much everyone.

**Comment by Gnarfoz** [comment]
It's endemic to modifying the map. The map frame probably contains secure buttons *somewhere* and modifying it from addons (i.e. insecure code) taints the frame, blocking secure code from running. It's a rabbit hole. :-\

**Comment by ynazar1** [comment]
Plumber is the only one i found that does something with map pins that *doesn't seem* to be having an issue. It puts seeds onto the emerald dream map.

**Comment by Shauren** [comment]
Weirdly enough this only causes the error to show up for me but I haven't spotted any broken functionality (yet)

**Comment by Shauren** [comment]
I wish blizzard didnt remove taintLog level 11 from release builds...

**Comment by GurliGebis** [comment]
Okay, I found the addon that does it.
I have `Kaliel's Tracker` installed, with this little checkbox enabled:
![billede](https://github.com/user-attachments/assets/f70de6c4-6ac1-48e7-abb6-e43fad309b13)

If I disable that checkbox, it starts failing like described here.

Here is a gist that contains the current code and the modified code when the checkbox is enabled: https://gist.github.com/GurliGebis/a7492ef3f2622ce721bce7a44856e527

@Shauren do you have any ideas? (Should we just replicate those changes?)

**Comment by GurliGebis** [comment]
The changes he does is:

```
    if newPin then
        pin:OnLoad();
        pin.UpdateMousePropagation = function() end <-- this has been added
    end
```
And this has been removed:
```
    -- Most pins should pass through right clicks to allow the map to zoom out
    -- This needs to be checked after OnAcquired because re-used pins can have
    -- dynamic setups that requires input propagation adjustment.
    pin:CheckMouseButtonPassthrough("RightButton");
```

**Comment by GurliGebis** [comment]
I have just released `11.0.2-20240910-2` - it is a large rewrite of the addon.
If you go into the options dialog, and enable the checkbox `Enable taint workarounds - Reloads UI (might have unknown effects)` - it should fix the issue.

Can you please give it a try and report back?

**Comment by Mike160770** [comment]
> I have just released `11.0.2-20240910-2` - it is a large rewrite of the addon. If you go into the options dialog, and enable the checkbox `Enable taint workarounds - Reloads UI (might have unknown effects)` - it should fix the issue.
> 
> Can you please give it a try and report back?

still happens - only a reload of the UI helps - but not for long

**Comment by xGux** [comment]
> I have just released `11.0.2-20240910-2` - it is a large rewrite of the addon. If you go into the options dialog, and enable the checkbox `Enable taint workarounds - Reloads UI (might have unknown effects)` - it should fix the issue.
> 
> Can you please give it a try and report back?

It still happens, specially in any WQ that needs to use an object.
When the issue happens a /reload fixes it until the next quest that uses an item.
Pretty much it breaks every single time in these quests.
![image](https://github.com/user-attachments/assets/37d970a8-ae93-4700-8965-c096670e66b8)

**Comment by GurliGebis** [comment]
Thanks, I'll take a look at it again, and see if I can figure out why it breaks

**Comment by GurliGebis** [comment]
I just tested it, with added print statements to verify the code is working.

Could I get you to try the following two things:
1: Disable all other addons.
2: Update the `Modules\Workarounds\WorkaroundsModule.lua` file, add the following line at the top of the `function WorldMapFrame:AcquirePin(pinTemplate, ...)` function, add the following line to the top of the function: `print("Function was called")`

Then reload the UI, open the map, and see if the chat is spammed with the text `Function was called`

**Comment by Mike160770** [comment]
Ok... I am waiting for the results from the other guy in this thread, because I am not familiar with rewriting add-ons.
Thx for your efforts, btw.

**Comment by GurliGebis** [comment]
Okay 🙂
The mentioned line is just to make it print something to the chat window, so we can verify that the code is actually being called.
So it's not that big of a change (if something goes wrong, you can already just reinstall the addon, and you are back to where you were before)

**Comment by GurliGebis** [comment]
Okay, so it is running - interesting.

Are you clicking on anything when it happens (like in the #67 issue) - just to see if that is the cause of it.

**Comment by xGux** [comment]
> Okay, so it is running - interesting.
> 
> Are you clicking on anything when it happens (like in the #67 issue) - just to see if that is the cause of it.

Yes, the error always shows up for me when clicking quest items from the Objective tracker.  
A lua error is captured in bugsack and an UI error saying "An addon has block an UI action" pops up on screen.
Reloading fixes it, until it happens again on the next quest

**Comment by GurliGebis** [comment]
I wonder how Kaliels Tracker does this, since it works with it.
I will have to look into it.

**Comment by GurliGebis** [comment]
@xGux do you happen to have a suggestion for a quest I can pick up and test with? (Right now, none of my characters have any quests with items in the quest tracker) 🙂

**Comment by Mike160770** [comment]
maybe the dailys in healing fall (sorry I play on a German client and do not know the English name for that zone)?

**Comment by GurliGebis** [comment]
Thanks, I'll take a look at it and see if I can track down what is causing it.

**Comment by ala75** [comment]
Something simular happends also when using the default interface to drop an item in for example a Delve. this can be 100% replacated for example in kriegers rest where you need to hold a candle for light. The moment you drop the light by clicking the "drop" button in the ui (Objective frame) you get an error i will post below. A UI reload fixes it for the session.

5x [ADDON_ACTION_FORBIDDEN] AddOn 'AngrierWorldQuests' tried to call the protected function 'CastSpellByID()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `CastSpellByID'
[string "@Blizzard_ObjectiveTracker/Blizzard_ScenarioObjectiveTracker.lua"]:1090: in function <...jectiveTracker/Blizzard_ScenarioObjectiveTracker.lua:1089>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

**Comment by GurliGebis** [comment]
This has been fixed in `11.0.2-20241003-2` - please test and reopen if the issue persists (remember to enable the taint workarounds in the options dialog)

And please report back, even if it works 😀

**Comment by GurliGebis** [comment]
That is really strange.
Can you try attaching a screenshot of your settings for the addon, so I can configure mine the same way?

**Comment by GurliGebis** [comment]
That works on my machine.
I cannot, no matter how hard I try, with just AWQ loaded and workarounds enabled, cause any errors.

Not sure what is happening for you, but without being able to reproduce it, it seems pretty impossible to fix (Since the errors mentioned indicate they are the issues already being worked around.

**Comment by GurliGebis** [comment]
Interresting.

Could I get you to install the `DebugLog` addon, enable it, and then replace the contents of the `WorkaroundsModule.lua` file with this: https://raw.githubusercontent.com/GurliGebis/WoWAddon-AngrierWorldQuests/refs/heads/debugging/Modules/Workarounds/WorkaroundsModule.lua ?
The changes I have done can be see here: https://github.com/GurliGebis/WoWAddon-AngrierWorldQuests/commit/688fa4ded6224dd1469c1602d7bc9a4eff0c51cf

Also, it might be an idea to install the two addons called `BugSack` and `BugGrabber` , since they make it easy to capture errors for pasting :)

Then, once you have done that, try and trigger the issue again - hopefully it should put a stacktrace inside the DebugLog, which should help pinpoint what is happening.

**Comment by GurliGebis** [comment]
That is the right spot, yes.
It is really strange - with the exact same addons loaded and configured identically - on the same version of the game client - there shouldn't be a possibility for different things to happen.
I'll keep looking into it - but I cannot promise anything yet.

**Comment by GurliGebis** [comment]
@dryya I just watched your video again.
I noticed when you log in, that you get the mission complete popup.

Can you try cleaning that out, so it doesn't popup, then log out and back in again, and then see if the problem goes away?

Would be interresting if that is the cause if it - but would be a pinpoint in the right direction.

**Comment by GurliGebis** [comment]
@dryya if you want, try and replace the contents of WorkaroundsModule.lua with this: https://raw.githubusercontent.com/GurliGebis/WoWAddon-AngrierWorldQuests/refs/heads/debugging/Modules/Workarounds/WorkaroundsModule.lua

Maybe that fixes it - it overrides the `SplashFrameMixin:OnHide()` function, and removes the call the update the objective tracker.

Please test and report back.

**Comment by Adrenalined** [comment]
> @dryya if you want, try and replace the contents of WorkaroundsModule.lua with this: https://raw.githubusercontent.com/GurliGebis/WoWAddon-AngrierWorldQuests/refs/heads/debugging/Modules/Workarounds/WorkaroundsModule.lua
> 
> Maybe that fixes it - it overrides the `SplashFrameMixin:OnHide()` function, and removes the call the update the objective tracker.
> 
> Please test and report back.

I swapped this file in and got this on the log on of a character I haven't been on in a while.

`1x ...orldQuests/Modules/Workarounds/WorkaroundsModule.lua:139: attempt to call global 'C_TalkingHead_SetConversationsDeferred' (a nil value)
[string "@AngrierWorldQuests/Modules/Workarounds/WorkaroundsModule.lua"]:139: in function <...orldQuests/Modules/Workarounds/WorkaroundsModule.lua:135>
[string "=[C]"]: in function `Hide'
[string "@Blizzard_UIParentPanelManager/Mainline/UIParentPanelManager.lua"]:901: in function `HideUIPanel'
[string "@Blizzard_FrameXML/SplashFrame.lua"]:94: in function `Close'
[string "*SplashFrame.xml:88_OnClick"]:1: in function <[string "*SplashFrame.xml:88_OnClick"]:1>

Locals:
frame = SplashFrame {
 BottomCloseButton = Button {
 }
 BottomTexture = Texture {
 }
 BottomLeftFeature = Frame {
 }
 showingQuestDialog = false
 RightFeature = Frame {
 }
 LeftTexture = Texture {
 }
 BottomLine = Texture {
 }
 TopLeftFeature = Frame {
 }
 Header = FontString {
 }
 RightTexture = Texture {
 }
 TopCloseButton = Button {
 }
 Label = FontString {
 }
}
fromGameMenu = false
(*temporary) = nil
(*temporary) = false
(*temporary) = "attempt to call global 'C_TalkingHead_SetConversationsDeferred' (a nil value)"`

**Comment by GurliGebis** [comment]
Good - now are are getting somewhere (even though it is broken).
@Adrenalined and @dryya I have updated the debugging branch - could you try with the code in here again: https://raw.githubusercontent.com/GurliGebis/WoWAddon-AngrierWorldQuests/refs/heads/debugging/Modules/Workarounds/WorkaroundsModule.lua

Now it overrides both `SplashFrameMixin:OnHide()` and `SplashFrameMixin:SetupFrame(screenInfo)`, removing the calls to `ObjectiveTrackerFrame:Update()` (from what I can see by searching to wow ui source code, we should now be handling every case where it happens)

Please test and report back 🙂

**Comment by Adrenalined** [comment]
Been running your updated file for most of the morning, seemed fine but this just popped.

I was in Hallowfall and I looted one of the purple outline treasure things on the ground.  On loot this happened.

1x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetPassThroughButtons'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:224: in function `CheckMouseButtonPassthrough'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:209: in function `AcquirePin'
[string "@Blizzard_SharedMapDataProviders/BonusObjectiveDataProvider.lua"]:97: in function <...haredMapDataProviders/BonusObjectiveDataProvider.lua:92>
[string "=[C]"]: in function `xpcall'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:76: in function `FireCallbacks'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:35: in function <.../Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua:31>
[string "=[C]"]: in function `accessor'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:52: in function `AddCallback'
[string "@Blizzard_ObjectAPI/Mainline/AsyncCallbackSystem.lua"]:60: in function `AddCancelableCallback'
[string "@Blizzard_SharedMapDataProviders/BonusObjectiveDataProvider.lua"]:92: in function `RefreshAllData'
[string "@Blizzard_SharedMapDataProviders/BonusObjectiveDataProvider.lua"]:65: in function `OnEvent'
[string "@Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua"]:99: in function `SignalEvent'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:106: in function <...aceBlizzard_MapCanvas/Blizzard_MapCanvas.lua:105>
[string "=[C]"]: in function `secureexecuterange'
[string "@Blizzard_MapCanvas/Blizzard_MapCanvas.lua"]:109: in function `OnEvent'
[string "@Blizzard_WorldMap/Blizzard_WorldMap.lua"]:148: in function <...rfaceBlizzard_WorldMap/Blizzard_WorldMap.lua:147>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}

**Comment by GurliGebis** [comment]
That is another issue. It seems like the AcquirePin function isn't being overridden like it should be.
I will provide an updated version that we can use to see if the AcquirePin function is still the original one, or the custom one.
Maybe later tonight, if I get the time.

@dryya do you still see your issue with the new changes?

**Comment by GurliGebis** [comment]
@prkr-ae did you enable the workarounds in settings?

**Comment by prkr-ae** [comment]
> @prkr-ae did you enable the workarounds in settings?

I haven't. I'll give it a try the next time I'm on

**Comment by prkr-ae** [comment]
> @prkr-ae did you enable the workarounds in settings?

ok, i turned the workarounds on. here's a couple more errors since then:

----1----

7x [ADDON_ACTION_FORBIDDEN] AddOn 'AngrierWorldQuests' tried to call the protected function 'CastSpellByID()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `CastSpellByID'
[string "@Blizzard_ObjectiveTracker/Blizzard_ScenarioObjectiveTracker.lua"]:1102: in function <...jectiveTracker/Blizzard_ScenarioObjectiveTracker.lua:1101>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

----2----

8x [ADDON_ACTION_FORBIDDEN] AddOn 'AngrierWorldQuests' tried to call the protected function 'UseQuestLogSpecialItem()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `UseQuestLogSpecialItem'
[string "@Blizzard_ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua"]:94: in function <...ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua:87>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

**Comment by martinboy1974** [comment]
I am also still getting lua error and I have enabled workaround

1x [ADDON_ACTION_FORBIDDEN] AddOn 'AngrierWorldQuests' tried to call the protected function 'UseQuestLogSpecialItem()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `UseQuestLogSpecialItem'
[string "@Blizzard_ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua"]:94: in function <...ObjectiveTracker/Blizzard_ObjectiveTrackerShared.lua:87>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

**Comment by martinboy1974** [comment]
> I'm looking into it - if you can give me steps you take to reproduce it, it would be a great help :)

The only thing that reproduces it, is accepting a quest that ads a use item in the quest tracker

**Comment by prkr-ae** [comment]
it's the same with world quests that give you an item to click in the quest tracker. you can go to the hallowfall spreading the light area and try with the one that gives you the item that reveals the hidden guys.

**Comment by jungheeMW** [comment]
Still getting this error in 2025/4/24
It occured when I finished a world quest
```
3x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.
[!BugGrabber/BugGrabber.lua]:485: in function <!BugGrabber/BugGrabber.lua:485>
[C]: in function 'SetPassThroughButtons'
[Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua]:280: in function 'CheckMouseButtonPassthrough'
[Blizzard_MapCanvas/Blizzard_MapCanvas.lua]:302: in function 'AcquirePin'
[Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:194: in function 'AddQuest'
[Blizzard_FlightMap/FM_QuestDataProvider.lua]:12: in function 'AddQuest'
[Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:134: in function 'CheckAddQuest'
[Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:141: in function 'RefreshAllData'
[Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:39: in function <...lizzard_SharedMapDataProviders/QuestDataProvider.lua:39>
[C]: ?
[Blizzard_SharedXMLBase/CallbackRegistry.lua]:144: in function <...eBlizzard_SharedXMLBase/CallbackRegistry.lua:143>
[C]: ?
[Blizzard_SharedXMLBase/CallbackRegistry.lua]:147: in function 'TriggerEvent'
[Blizzard_FrameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua]:59: in function 'CacheCurrentSuperTrackInfo'
[Blizzard_FrameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua]:17: in function <...rameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua:15>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
```

**Comment by GurliGebis** [comment]
> Still getting this error in 2025/4/24 It occured when I finished a world quest
> 
> ```
> 3x [ADDON_ACTION_BLOCKED] AddOn 'AngrierWorldQuests' tried to call the protected function 'Button:SetPassThroughButtons()'.
> [!BugGrabber/BugGrabber.lua]:485: in function <!BugGrabber/BugGrabber.lua:485>
> [C]: in function 'SetPassThroughButtons'
> [Blizzard_MapCanvas/MapCanvas_DataProviderBase.lua]:280: in function 'CheckMouseButtonPassthrough'
> [Blizzard_MapCanvas/Blizzard_MapCanvas.lua]:302: in function 'AcquirePin'
> [Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:194: in function 'AddQuest'
> [Blizzard_FlightMap/FM_QuestDataProvider.lua]:12: in function 'AddQuest'
> [Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:134: in function 'CheckAddQuest'
> [Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:141: in function 'RefreshAllData'
> [Blizzard_SharedMapDataProviders/QuestDataProvider.lua]:39: in function <...lizzard_SharedMapDataProviders/QuestDataProvider.lua:39>
> [C]: ?
> [Blizzard_SharedXMLBase/CallbackRegistry.lua]:144: in function <...eBlizzard_SharedXMLBase/CallbackRegistry.lua:143>
> [C]: ?
> [Blizzard_SharedXMLBase/CallbackRegistry.lua]:147: in function 'TriggerEvent'
> [Blizzard_FrameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua]:59: in function 'CacheCurrentSuperTrackInfo'
> [Blizzard_FrameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua]:17: in function <...rameXMLUtil/Mainline/Blizzard_QuestSuperTracking.lua:15>
> 
> Locals:
> _ = Frame {
> }
> event = "ADDON_ACTION_BLOCKED"
> events = <table> {
> }
> ```

Did you enable the workarounds in the addon settings?

**Comment by jungheeMW** [comment]
I believe so. Here are my settings if you would like to check.

<img width="466" alt="Image" src="https://github.com/user-attachments/assets/7eef8587-2992-42df-8ef0-73edea4af4bc" />

**Comment by jungheeMW** [comment]
Even with my full addon suite the error cannot be reproduced reliably. 
However, reading through the above comments, they mentioned the HereBeDragons API, which I think is used by Mapster, HandyNotes, and maybe SilverDragon?

I'll try running without those addons for a while and report back if I get the error to occur.

**Comment by GurliGebis** [comment]
Okay 😊
